### DISCUSSION TITLE
Use opengl instead of vulkan

### DISCUSSION THREAD

**Discussion by theoparis** [zed-industries/zed#12759] [Feature Requests]
How can I get the https://github.com/zed-industries/zed/releases/tag/v0.139.1-pre on Linux to use opengl instead of vulkan? This laptop I have is a T420 with a 2nd gen intel cpu and does not support vulkan, therefore everything on Zed is laggy because of lavapipe.

Edit: related? https://github.com/zed-industries/zed/issues/13305

**Comment by Moshyfawn** [comment]
Check out this discussion.

https://github.com/zed-industries/zed/issues/8168#issuecomment-2116604666

**Reply by theoparis** [reply]
I don't see where it mentions running opengl with the prerelease binaries. Do I have to cross compile zed to add opengl support?

**Reply by theoparis** [reply]
Update: I built zed on another Linux system using RUSTFLAGS="--cfg gles" and I got this at runtime when running zed:
```
Thread "main" panicked with "called `Result::unwrap()` on an `Err` value: MissingFeatures(Features(BUFFER_STORAGE | DYNAMIC_ARRAY_SIZE))" at /root/.cargo/git/checkouts/blade-ea462a0faa3f9995/33fd513/blade-graphics/src/gles/pipeline.rs:142:14
```

It seems like blade assumes OpenGL 4.6. See https://github.com/kvark/blade/issues/163

**Comment by LimaneGaya** [comment]
Same issue here, does zed only support vulkan?

**Comment by stalkerg** [comment]
For idea behind Zed so old hardware will not work anyway. I think for really old or slow hardware will be better to use something like neovim. 
Also soon binary distros will switch to V2 or V3 (optimization profile) and you CPU will not support anyway.

**Reply by uhwhatamityping** [reply]
It worked on my ancient gt 220. Now i just need to figure out how to get it to work in linux

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support

**Comment by z3r0n3br4instorm** [comment]
Hi Everyone, I am maintaining this OpenGL3.3 based fork of zeditor for my MacBook Pro 2011 with HD3000 Graphics. Here's the link if anyone wanna check it out https://github.com/z3r0n3br4instorm/zed/tree/gl33-backend

**Reply by theoparis** [reply]
~~Is this even needed anymore now that Zed uses WGPU?~~ Ah, nvm as you pointed out in the commit it requires certain features that the old mbp doesn't support. It would be nice if it would be able to be upstreamed to GPUI itself 😄

**Reply by z3r0n3br4instorm** [reply]
Yeah, Let's see. Also T420 has the CPU Gen as mine i think, I have a pre-built binary if you want.

https://drive.google.com/file/d/1BwfKQZ244XbmhXzCnXNBHm9PPlfuZTIr/view?usp=sharing
